hoodie-common/src/main/java/com/uber/hoodie/common/model/HoodieFileGroup.java

```diff
@@ -141,15 +165,29 @@ private boolean isFileSliceCommitted(FileSlice slice) {
   public Optional<FileSlice> getLatestFileSliceBeforeOrOn(String maxCommitTime) {
     return getAllFileSlices()
         .filter(slice ->
-            HoodieTimeline.compareTimestamps(slice.getBaseCommitTime(),
+            HoodieTimeline.compareTimestamps(slice.getBaseInstantTime(),
```

Comment:
As discussed we need to return the penultimate data file.. this method will only return the lastSlice (pending compaction) file and that can be a problem.

Reply:
@vinothchandar : I have updated all the APIs (in FileSystemView) returning DataFiles to ensure the correct data-file (penultimate one) gets picked. For all the APIs (in FileSystemView) returning file-slices (for RT view), the existing APIs are made to return only the last file-slice (not the penultimate one). For use-cases that require the merged/penultimate file-slices (e.g: MergeOnRead table), the new API getLatestMergedFileSlicesBeforeOrOn is used. BloomIndex uses one of the DataFiles API in FileSystemView (getLatestDataFilesBeforeOrOn)